src/main/java/net/datafaker/providers/base/Compass.java

```diff
@@ -0,0 +1,110 @@
+package net.datafaker.providers.base;
+
+/**
+ * @since 1.7.0
+ */
+public class Compass extends AbstractProvider<BaseProviders> {
+
+    enum CompassPoint { CARDINAL, ORDINAL, HALF_WIND, QUARTER_WIND }
+
+    private CompassPoint compassPoint;
+
+    protected Compass(BaseProviders faker) {
+        super(faker);
+    }
+
+    /**
+     * Specify point of direction. If not specified, point of direction will be randomly selected.
+     * @param compassPoint point of direction.
+     * @return itself with specified point of direction.
+     * @see CompassPoint
+     */
+    public Compass compassPoint(CompassPoint compassPoint) {
+        this.compassPoint = compassPoint;
+        return this;
+    }
+
+    /**
+     * Returns full-size name of direction according to the specified {@link CompassPoint} point of direction.
+     * <p>
+     * Specify point of direction using {@link Compass#compassPoint(CompassPoint)} method.
+     * <p>
+     * Typical call of this method: {@code compass().compassPoint(CompassPoint.CARDINAL).word();}
+     *
+     * @return the full-size direction word according to the specified {@link CompassPoint}. Otherwise, the point of direction will be selected randomly.
+     */
+    public String word() {
+        if (compassPoint == null) {
+            return resolve("compass.direction");
+        }
+
+        switch (compassPoint) {
+            case CARDINAL:
+                return resolve("compass.cardinal.word");
```

Comment:
I think it would be better to get remove these switch statements, and create the resolve string by templating this string, something like:

`"compass." + compassPoint.key + ".word`

Reply:
Fixed

Reply:
Nice. But now the default fallback you had before is gone. Was that not needed?

Reply:
This is not necessary because each method checks for null, and if the user has not specified a direction, the null check should work and use a random direction